packages/aft/lib/src/commands/generate/generate_amplify_swift_command.dart

```diff
@@ -0,0 +1,288 @@
+            'https://github.com/aws-amplify/amplify-swift.git',
+            cloneDir.path,
+          ],
+          echoOutput: verbose,
+        );
+        logger.info('Successfully cloned Amplify Swift Repo');
+        return cloneDir;
+      });
+
+  /// Checks out [ref] in [pluginDir].
+  Future<Directory> _checkoutRepositoryRef(
+    Directory pluginDir,
+    String ref,
+  ) async {
+    logger
+      ..info('Checking out target branch: $ref')
+      ..verbose('Creating git work tree in $pluginDir');
+    final worktreeDir =
+        await Directory.systemTemp.createTemp('amplify_swift_worktree_');
+    try {
+      await runGit(
+        ['worktree', 'add', worktreeDir.path, ref],
+        processWorkingDir: pluginDir.path,
+      );
+    } on Exception catch (e) {
+      if (e.toString().contains('already checked out')) {
+        return pluginDir;
+      }
+      rethrow;
+    }
+    return worktreeDir;
+  }
+
+  /// Sets up the Amplify Swift repo for use later
+  Future<void> _setupRepo() async {
+    if (_repoCache[branchTarget] != null) {
+      return;
+    }
+    Directory repoDir;
+    repoDir = await _downloadRepository();
+    repoDir = await _checkoutRepositoryRef(repoDir, branchTarget);
+
+    _repoCache[branchTarget] = repoDir;
+  }
+
+  /// Returns the directory for the plugin at [path].
+  Future<Directory> _pluginDirForPath(String path) async {
+    final repoDir = _repoCache[branchTarget];
+    if (repoDir == null) {
+      exitError('No cached repo for branch $branchTarget');
+    }
+
+    return Directory.fromUri(
+      repoDir.uri.resolve(path),
+    );
+  }
+
+  /// Generates the Amplify Swift plugin for [plugin].
+  Future<void> _generatePlugin(PluginConfig plugin) async {
+    logger.info('Selecting source files for ${plugin.name}...');
+
+    final repoDir = await _pluginDirForPath(plugin.remotePathToSource);
+
+    final outputDir = Directory(p.join(outputPath, plugin.name));
+
+    // Clear out the directory if it already exists.
+    // This is to ensure that we don't have any stale files.
+    if (await outputDir.exists()) {
+      logger.verbose(
+        'Clearing out existing plugin directory for ${plugin.name}...',
+      );
+      await _setPermissions(outputDir.path, PermissionScope.readExecuteWrite);
+      await outputDir.delete(recursive: true);
+    }
+    await outputDir.create(recursive: true);
+
+    // Copy the files from the repo to the plugin directory.
+    logger
+      ..info('Copying plugin files for ${plugin.name}...')
+      ..verbose('From $repoDir to $outputDir');
+    await copyPath(repoDir.path, outputDir.path);
+    await _setPermissions(outputDir.path, PermissionScope.readExecute);
+  }
+
+  Future<void> checkDiff(PluginConfig plugin) async {
+    logger.info('Checking diff for ${plugin.name}...');
+    final incoming = (await _pluginDirForPath(plugin.remotePathToSource)).path;
+    // Set the permissions of the incoming files to readExecute (fails otherwise)
```

Comment:
Q: What fails when the permissions are not updated?

Reply:
diff command catches the literal difference of permissions. Ie, 555 local vs 755 remote.